### PR TITLE
fix(useDeprecatedSynchronousErrorHandling): now throws properly when notifying a Subject

### LINKED ISSUE #6471: Super Gross Mode @ Google: Subjects no longer throw while nexting
version: 7.x

```ts
config.useDeprecatedSynchronousErrorHandling = true;

const subject = new Subject();

const source = subject.pipe(switchMap(() => throwError('bad')));
source.subscribe(console.log);

try {
  subject.next('bha');
} catch (err) {
  console.log('here');
}
```

cc @leggechr

### PR DESCRIPTION
**Pull request by benlesh**
Resolves #6471

### CODE REVIEW

**Comment by benlesh** [comment]
cc @leggechr

File: src/internal/Subscriber.ts [hunk 1]
```diff
@@ -84,6 +84,9 @@ export class Subscriber<T> extends Subscription implements Observer<T> {
    */
   error(err?: any): void {
     if (this.isStopped) {
+      if (config.useDeprecatedSynchronousErrorHandling) {
+        throw err;
+      }
```
Comment (benlesh) [inline src/internal/Subscriber.ts:89]: @leggechr the fix is this bit here.

File: src/internal/Subscriber.ts [hunk 2]
```diff
@@ -83,7 +83,7 @@ export class Subscriber<T> extends Subscription implements Observer<T> {
    * @return {void}
    */
   error(err?: any): void {
-    if (this.isStopped) {
+    if (this.isStopped && !config.useDeprecatedSynchronousErrorHandling) {
```
Comment (benlesh) [inline src/internal/Subscriber.ts:86]: @leggechr this is the fix.

File: api_guard/dist/types/index.d.ts [hunk 3]
```diff
@@ -249,7 +249,7 @@ export interface ObjectUnsubscribedError extends Error {
 
 export declare const ObjectUnsubscribedError: ObjectUnsubscribedErrorCtor;
 
-export declare const observable: string | symbol;
+export declare const observable: string | typeof Symbol.observable;
```
Comment (cartant) [inline api_guard/dist/types/index.d.ts:253]: Why has this changed? The last time this happened, IIRC, it was in that PR that sought to bump to TS 4.3. Are you sure you have 4.2 installed?

File: src/internal/Subscriber.ts (hunk 2 above)
Comment (cartant) [reply src/internal/Subscriber.ts:86]: Surely this is going to cause other problems? This renders stopped meaningless for error notifications if that config is used? How did this ever work in v6? If it didn't, should we really change this? FWIW, I don't particularly care, but I thought I'd point it out.

**Review by cartant — APPROVED** [review]

**Review by cartant — CHANGES_REQUESTED** [review]
Actually, the api_guardian thing needs to be addressed.

**Comment by leggechr** [comment]
This caused tons more failing tests in google. The error is: `TypeError: Cannot read property 'error' of null` at this line: https://github.com/ReactiveX/rxjs/blob/master/src/internal/Subscriber.ts#L123

**Comment by benlesh** [comment]
Completely redone... very hacky for now... only one issue left though (marked with `it.skip` in the tests). cc @cartant @leggechr

**Comment by benlesh** [comment]
Closing in favor of #6478

**Pull request closed without merging**